internal/cmd/perfcomp/energystatistics_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) MongoDB, Inc. 2025-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gonum.org/v1/gonum/mat"
+)
+
+func createTestVectors(start1 int, stop1 int, step1 int, start2 int, stop2 int, step2 int) (*mat.Dense, *mat.Dense) {
+	xData := []float64{}
+	yData := []float64{}
+
+	i := start1
+	for i < stop1 {
+		xData = append(xData, float64(i))
+		i += step1
+	}
+	j := start2
+	for j < stop2 {
+		yData = append(yData, float64(j))
+		j += step2
+	}
+
+	x := mat.NewDense(len(xData), 1, xData)
+	y := mat.NewDense(len(yData), 1, yData)
+
+	return x, y
+}
+
+func TestEnergyStatistics(t *testing.T) {
+
+	t.Run("similar distributions should have small e,t,h values ", func(t *testing.T) {
+		x, y := createTestVectors(1, 100, 1, 1, 105, 1)
+		e, tstat, h := GetEnergyStatistics(x, y)
+
+		del := 1e-3
+
+		assert.InDelta(t, 0.160, e, del) // |0.160 - e| < 1/100
+		assert.InDelta(t, 8.136, tstat, del)
+		assert.InDelta(t, 0.002, h, del)
+	})
+
+	t.Run("different distributions should have large e,t,h values", func(t *testing.T) {
+		x, y := createTestVectors(1, 100, 1, 10000, 13000, 14)
+		e, tstat, h := GetEnergyStatistics(x, y)
+		del := 1e-3
+
+		assert.InDelta(t, 21859.691, e, del)
+		assert.InDelta(t, 1481794.709, tstat, del)
+		assert.InDelta(t, 0.954, h, del)
+	})
+
+	t.Run("uni-variate distributions", func(t *testing.T) {
+		x, y := createTestVectors(1, 300, 1, 1000, 5000, 10)
+		e, tstat, h := GetEnergyStatistics(x, y)
+		del := 1e-3
+
+		assert.InDelta(t, 4257.009, e, del)
+		assert.InDelta(t, 728381.015, tstat, del)
+		assert.InDelta(t, 0.748, h, del)
+	})
+
+	t.Run("equal distributions should have all 0 values", func(t *testing.T) {
+		x := mat.NewDense(10, 1, []float64{1, 1, 1, 1, 1, 1, 1, 1, 1, 1})
+		y := mat.NewDense(1, 1, []float64{1})
+
+		e, tstat, h := GetEnergyStatistics(x, y)
+		del := 1e-3
+
+		assert.InDelta(t, 0.0, e, del)
+		assert.InDelta(t, 0.0, tstat, del)
+		assert.InDelta(t, 0.0, h, del)
```

Comment:
Can these be `Equal` instead of `InDelta`?


```suggestion
		assert.Equal(t, 0.0, e)
		assert.Equal(t, 0.0, tstat)
		assert.Equal(t, 0.0, h)
```